src/itwinai/torch/loggers.py

```diff
@@ -0,0 +1,238 @@
+
+import os
+import tempfile
+from argparse import Namespace
+from pathlib import Path
+from typing import Any, Dict, Literal, Optional, Union
+
+import yaml
+from lightning.pytorch.callbacks.model_checkpoint import ModelCheckpoint
+from lightning.pytorch.loggers.logger import Logger as LightningLogger
+from lightning.pytorch.loggers.logger import rank_zero_experiment
+from lightning.pytorch.loggers.utilities import _scan_checkpoints
+from lightning.pytorch.utilities import rank_zero_only
+from torch import Tensor
+from typing_extensions import override
+
+from itwinai.loggers import Logger
+
+
+class PyTorchLightningLogger(LightningLogger):
+    """Adapter between PyTorch Lightning logger and itwinai logger.
+
+    This adapter forwards logging calls from PyTorch Lightning to the
+    itwinai Logger instance, using the itwinai Logger's `log` method.
+    It supports the lightning logging of metrics, hyperparameters, and checkpoints.
+    Additionally, any function calls can be forwarded to the itwinai logger instance
+    though the `experiment` property of this Adapter.
+    """
+
+    def __init__(
+        self,
+        itwinai_logger: Logger,
+        log_model: Union[Literal["all"], bool] = False
+    ):
+        """Initializes the adapter with an itwinai logger instance.
+
+        Args:
+            itwinai_logger (Logger): An instance of itwinai Logger.
+            log_model (Union[Literal["all"], bool], optional):
+                Specifies which checkpoints to log.
+                If "all", logs all checkpoints; if True, logs the best k checkpoints according
+                to the specifications given as `save_top_k` in the Lightning ModelCheckpoint;
+                if False, does not log checkpoints.
+        """
+        self.itwinai_logger = itwinai_logger
+
+        self._log_model = log_model
+        self._logged_model_time = {}
+        self._checkpoint_callback = None
+
+        self._initialized = False
+
+    @property
+    def name(self) -> Optional[str]:
+        """Return the experiment name."""
+
+    @property
+    def version(self) -> Optional[Union[int, str]]:
+        """Return the experiment version."""
+
+    @property
+    @override
+    @rank_zero_only
+    def save_dir(self) -> Optional[str]:
+        return self.experiment.savedir
+
+    @property
+    @rank_zero_experiment
+    def experiment(self) -> Logger:
+        """Lightning Logger function.
+        Initializes and returns the itwinai Logger context for experiment tracking.
+
+        Returns:
+            Logger: The itwinai logger instance.
+        """
+        if not self._initialized:
+            # With the rank_zero_decorator the rank will always be 0
+            self.itwinai_logger.create_logger_context(rank=0)
+            self._initialized = True
+
+        return self.itwinai_logger
+
+    @override
+    @rank_zero_only
+    def finalize(
+        self,
+        status: str
```

Comment:
I can see that `status` is unused, but you need it because it is also defined in the super class' signature. I would note this in the docstring, maybe even instead of what it currently says, and potentially set a default argument so that you don't have to pass it in. Not sure how well that works with other functions calling it, but should be fine?